File: batch/src/main/java/com/isystk/sample/batch/jobs/solrRegist/SolrRegistTasklet.java
package com.isystk.sample.batch.jobs.solrRegist;

import com.isystk.sample.batch.context.BatchContext;
import com.isystk.sample.batch.jobs.BaseTasklet;
import com.isystk.sample.batch.service.SolrPostService;
import java.io.IOException;
import org.slf4j.Logger;
import org.springframework.batch.core.StepContribution;
import org.springframework.batch.core.scope.context.ChunkContext;
import org.springframework.batch.repeat.RepeatStatus;
import org.springframework.beans.factory.annotation.Autowired;

/**
 * インデックス生成
 */
public class SolrRegistTasklet extends BaseTasklet<SolrRegistPostDto> {

  private static final Logger log = org.slf4j.LoggerFactory.getLogger(SolrRegistTasklet.class);
  @Autowired
  SolrPostService solrPostService;

  @Override
  public RepeatStatus execute(StepContribution contribution, ChunkContext chunkContext)
      throws IOException {
    return super.execute(contribution, chunkContext);
  }

  @Override
  protected void doProcess(BatchContext context) {
    // Solrの投稿インデックスを更新します。
    solrPostService.refresh();
  }

}

File: batch/src/main/kotlin/com/isystk/sample/batch/jobs/solrRegist/SolrRegistTasklet.kt
package com.isystk.sample.batch.jobs.solrRegist

import com.isystk.sample.batch.context.BatchContext
import com.isystk.sample.batch.jobs.BaseTasklet
import com.isystk.sample.batch.service.SolrStockService
import org.springframework.batch.core.StepContribution
import org.springframework.batch.core.scope.context.ChunkContext
import org.springframework.batch.repeat.RepeatStatus
import org.springframework.beans.factory.annotation.Autowired
import java.io.IOException

/**
 * インデックス生成
 */
class SolrRegistTasklet : BaseTasklet<SolrRegistDto>() {
    @Autowired
    var solrStockService: SolrStockService? = null
    @Throws(IOException::class)
    override fun execute(contribution: StepContribution, chunkContext: ChunkContext): RepeatStatus? {
        return super.execute(contribution, chunkContext)
    }

    override fun doProcess(context: BatchContext?) {
        // Solrの投稿インデックスを更新します。
        solrStockService!!.refresh()
    }

}